extensions/catalog/catalog-atlas/src/main/java/com/microsoft/dagx/catalog/atlas/metadata/AtlasMetadataStore.java

```diff
@@ -0,0 +1,134 @@
+import com.microsoft.dagx.spi.types.domain.transfer.DataAddress;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.toSet;
+
+@SuppressWarnings("unchecked")
+public class AtlasMetadataStore implements MetadataStore {
+    private static final String ATLAS_PROPERTY_KEYNAME = "keyName";
+    private static final String ATLAS_PROPERTY_TYPE = "type";
+    private final AtlasApi atlasApi;
+    private final Monitor monitor;
+
+    public AtlasMetadataStore(AtlasApi atlasApi, Monitor monitor) {
+        this.atlasApi = atlasApi;
+        this.monitor = monitor;
+    }
+
+    @Override
+    public @Nullable DataEntry<?> findForId(String id) {
+
+        var properties = atlasApi.getEntityById(id);
+        if (properties == null) {
+            var searchResult = atlasApi.dslSearchWithParams("from AzureStorage where name = '" + id + "'", 100, 0);
+            properties = searchResult.getEntities().stream()
+                    .filter(eh -> eh.getStatus() == AtlasEntity.Status.ACTIVE)
+                    .map(eh -> atlasApi.getEntityById(eh.getGuid())).findFirst().orElse(null);
+            if (properties == null) {
+                return null;
+            }
+        }
+        final AtlasEntity entity = properties.getEntity();
+        if (entity == null) {
+            monitor.info("AtlasMetadataStore: no DataEntry found for ID " + id);
+            return null;
+        }
+        var policyId = getPolicyIdForEntity(entity);
+        var address = DataAddress.Builder.newInstance()
+                .keyName(entity.getAttribute(ATLAS_PROPERTY_KEYNAME).toString())
+                .type(entity.getAttribute(ATLAS_PROPERTY_TYPE).toString())
+                .properties(convert(entity.getAttributes()))
+                .build();
+
+
+        return DataEntry.Builder.newInstance()
+                .id(StringUtils.toString(entity.getAttribute("name")))
+                .policyId(policyId)
+                .catalog(new AtlasDataCatalog(address))
+                .build();
+
+    }
+
+    private String getPolicyIdForEntity(AtlasEntity entry) {
+
+        if (entry == null) {
+            return null;
+        }
+        Map<String, String> policyProperties = (Map<String, String>) entry.getRelationshipAttribute("itsAccessPolicy");
+        if (policyProperties == null) {
+            return null;
+        }
+        final String policyGuid = policyProperties.get("guid");
+
+        //lets make sure the policy actually exists
+        final AtlasEntity.AtlasEntityWithExtInfo policyEntity = atlasApi.getEntityById(policyGuid);
+        return policyEntity != null ? StringUtils.toString(policyEntity.getEntity().getAttribute("policyId")) : null;
+    }
+
+    @Override
+    public void save(DataEntry<?> entry) {
+        monitor.severe("Save not yet implemented");
```

Comment:
Is there a reason you couldn't implement save yet?

Reply:
mostly because of two things: 
1) inserting data into the catalog should (in most cases) happen out of band (e.g. through seeding, batch jobs, etc.)
2) A mapping from `DataAddress` to the Atlas-specific entity type definition would be needed, which is as of yet unspecified/unimplemented. For the E2E-flow that is not necessary though.